uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/WorkerHelixManager.java

```diff
@@ -184,6 +184,25 @@ public synchronized void addTopicToMirrorMaker(InstanceTopicPartitionHolder cont
     }
   }
 
+  public synchronized void expandWorker(InstanceTopicPartitionHolder controller, String pipeline, int routeId){
+    TopicPartition route = new TopicPartition(pipeline, routeId);
+    List<String> instances = _routeToInstanceMap.get(route);
+    List<String> addInstances = new ArrayList<>();
+    if (_availableWorkerList.size() > 0) {
+      int currentWorkerSize = instances.size();
+      for (int i = 0; i < _availableWorkerList.size() && currentWorkerSize <= _conf.getInitMaxNumWorkersPerRoute(); i++, currentWorkerSize++) {
+        instances.add(_availableWorkerList.get(i));
```

Comment:
why add more workers in here? 
Process will assign workers to uReplicator route on route creation. worker number it should be sufficient, number of 
 worker rescale should be based on workload instead of num of topics.

Reply:
when assign workers to uReplicator route on route creation, the available workers count is less than getInitMaxNumWorkersPerRoute, it will assign current available workers size to the route 。 then laster some time，we have more available workers (for example, starting more or etc), and we add topic to uReplicator, we need to add more worker 。